internal/tools/changes/release.go

```diff
@@ -0,0 +1,69 @@
+package changes
+
+import (
+	"fmt"
+	"sort"
+)
+
+// VersionBump describes a version increment to a module.
+type VersionBump struct {
+	From string
+	To   string
+}
+
+// Release represents a single SDK release, which contains all change metadata and their resulting version bumps.
+type Release struct {
+	ID            string
+	SchemaVersion string
+	VersionBumps  map[string]VersionBump
+	Changes       []*Change
+}
+
+// RenderChangelogForModule returns a new markdown section of a module's CHANGELOG based on the Changes in the Release.
+func (r *Release) RenderChangelogForModule(module, headerPrefix string) string {
+	sections := map[string]string{}
+	var sectionsList []string
+
+	for _, c := range r.Changes {
+		if c.Module == module {
+			if _, ok := sections[c.Type]; !ok {
+				sectionsList = append(sectionsList, c.Type)
+			}
+
+			sections[c.Type] += fmt.Sprintf("* %s\n", c.Description)
+		}
+	}
+
+	entry := fmt.Sprintf("%s# `%s`", headerPrefix, module)
+	if bump, ok := r.VersionBumps[module]; ok {
+		entry += " - " + bump.To + "\n"
+	} else {
+		entry += "\n"
+	}
+
+	sort.Strings(sectionsList)
+
+	for _, section := range sectionsList {
+		entry += headerPrefix + "## " + changeHeaders[section] + "\n"
+		entry += sections[section] + "\n"
+	}
+
+	return entry
+}
```

Comment:
Might be easier to follow and write to investigating using [text/template#Template](https://pkg.go.dev/text/template?tab=doc) for the changelog. It also allows easier formatting that can be viewed at a glance.

Reply:
I switched this function over to using a `text/template`. It's still not the most clear thing to look at, but I think it's definitely better at a glance than the original function.